docs/doc/dev/doc-generation/how-to.adoc

```diff
@@ -0,0 +1,81 @@
+= Generate rs-server technical documentation
+
+This procedure is run automatically in the CI by the 'generate documentation' workflow.
+It can be executed locally to verify the generated documentation (before publishing it for example).
+
+See ${include("description.adoc")} for more details on the process and technical stuff.
+
+. Pre-requises
+
+The local developer environment has been setup.
+The global python project is setup.
+The python sub-project for doc generation is also setup.
+
+. Generate the technical documentation
+
+From the rs-server/docs folder, execute the command:
+[source, bash, indent=0]
+----
+poetry run asciidoxy --base-dir doc --build-dir ../dist/doc/ --multipage doc/index.adoc
+----
+
+This command generates the html pages of all the technical documentation
+and write them in the rs-server/dist/doc folder.
+
+. Generate the python api documentation
+
+From the root folder, run the following command :
+[source, bash, indent=0]
+----
+poetry run sphinx-apidoc -o docs/doc/api/python/generated rs_server
+----
+It generates rst files from the python doc-strings
+and write them in a generated folder (excluded of git conf).
+
+Then, execute the following command :
+[source, bash, indent=0]
+----
+poetry run sphinx-build -M html docs/doc/api/python/ dist/doc/output/api/python/
+----
+It generates the html pages for the python api of the rs-server.
+The generated api is accessible locally at 'dist/doc/api/html/index.html'.
+
+. Generate the REST API documentation
+
+From the root folder, run the following command :
+[source, bash, indent=0]
+----
+mkdir -p dist/doc/output/api/rest/
+cp docs/doc/api/rest/index.html dist/doc/output/api/rest/
+----
+
+The previous command copies the entry document of the REST API
+to the folder in which the documentation is generated.
+It's important to keep it like that for integration with the technical documentation.
+
+From the root folder, run the following command :
+
+[source, bash, indent=0]
+----
+poetry run python -m rs_server.CADIP.openapi dist/doc/output/api/rest/
+----
+
+The previous command uses FastAPI to extract the openapi specification of the rs-server CADIP service and write it on the expected folder.
+The output folder is the one expected for integration with the main document of the rest api previously copied.
+
+NOTE: This is a POC waiting for the frontend to be implemented.
+The documented REST API will be the one of the frontend at the end.
+
+. Verify the generated documentation
+
+The main page of the technical documentation is accessible
+http://localhost:63342/rs-server/dist/doc/output/index.html[locally].
```

Comment:
What is the service that runs on 63342 necessary to see the doc ?

If a open the local index.html file I have error "Failed to fetch openapi.json"

Reply:
It's a service provided by Pycharm.
I removed this from the documentation.
I added a note to explain that the swagger documentation cant be verified locally
if not served.